### PR TITLE
Improve README clarity and technical focus

### PR DESCRIPTION
**Pull request by nicolai-rhesis**
## Purpose
Address community feedback to make the README more developer-focused and clearly communicate what Rhesis does, who it's for, and how it differentiates from alternatives.

## What Changed
- **Clearer opening**: Replaced generic 'testing platform' language with specific description of what Rhesis does (generates test inputs, evaluates outputs)
- **Added differentiation section**: Explicitly compares Rhesis to manual testing, traditional frameworks, observability tools, and red-teaming services
- **Removed marketing buzzwords**: Eliminated terms like 'collaborative test management', 'unlimited user creativity', 'hand-coded scenarios can't scale'
- **Added 'How Rhesis Works' section**: 4-step explanation of the actual workflow
- **Highlighted key differentiators**:
  - Single-turn and multi-turn testing (Penelope agent for conversations)
  - Built for cross-functional teams (UI for non-tech, SDK for engineers, collaborative features)
  - Support for LLM and agentic applications (chatbots, RAG, agents with tool use)
  - Comprehensive metrics library from popular frameworks (RAGAS, DeepEval)
  - SDK for IDE workflow
- **Updated local Docker setup**: Added accurate details about ./rh start command and what it does automatically
- **Streamlined sections**: Made Contributing, Support, and License sections more concise
- **Developer-focused language**: Replaced enterprise CEO pitch with practical technical value

## Additional Context
- Based on community feedback about README being too marketing-heavy
- Maintains all existing functionality while improving communication
- No code changes, documentation only

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -39,46 +39,56 @@ Rhesis is an open-source testing platform that transforms how Gen AI teams valid
...
+You get a test suite that covers edge cases you wouldn't have thought of, runs automatically, and shows exactly where your LLM fails.
 
-**Your whole team should define what matters**
+## What Makes This Different
 
-Your legal, marketing, and domain experts know what can actually go wrong. Rhesis makes testing everyone's responsibility.
+**Single-turn and multi-turn testing**: Test both simple Q&A and complex conversations. Penelope (our multi-turn agent) simulates realistic user conversations with multiple back-and-forth exchanges to catch issues that only appear in extended interactions. Works with chatbots, RAG systems, and agentic applications.
 
-**Know what you're shipping**
+**Built for teams, not just engineers**: UI for non-technical stakeholders to define requirements and review results. SDK for engineers to work from their IDE and integrate into CI/CD. Comments, tasks, and review workflows so legal, compliance, and domain experts can collaborate without writing code.
```
Comment (alexeyhramov123) [inline README.md:61]: Comments, tasks, and review workflows so legal, compliance, and domain experts can collaborate without writing code. - this sentence is hard to read. maybe:
Legal, compliance and domain experts can collaborate using comments, tasks, and review workflows without writing code.
